### ISSUE TITLE
helm template emits duplication into secrets.yaml for ADVERTISED_KAFKA_ADDRESSES and other shell vars

### ISSUE THREAD

**Issue by hcoyote** [redpanda-data/helm-charts#1341]
### What happened?

This is more for future tracking and cleanup. 

`helm template` emits some duplication in the secrets.yaml along the lines of multiple ADVERTISED_KAFKA_ADDRESSES and ADVERTISED_HTTP_ADDRESSES that would end up dropping the same values multiple times into the shell arrays.

It doesn't seem to specifically affect anything that we can see, but it's created questions from customers, so logging this more for future cleanup rather than it being a specific bug that seems to be causing issues.

The following is what gets repeated multiple times.

```
    PREFIX_TEMPLATE=""
    ADVERTISED_KAFKA_ADDRESSES+=("{\"address\":\"${SERVICE_NAME}.redpanda.rpnamespace.example.net\",\"name\":\"default\",\"port\":9092}")
```
and
```
    PREFIX_TEMPLATE=""
    ADVERTISED_HTTP_ADDRESSES+=("{\"address\":\"${SERVICE_NAME}.redpanda.rpnamespace.example.net\",\"name\":\"default\",\"port\":30082}")

```

Per @chrisseto from internal discussion:

> that's just an unfortunate part of how the chart works right now. It can look different if you're using a certain feature set (Setting prefix template on the listeners to have more than 1 value) but it falls back to behavior that looks pretty buggy at first glance.
> 
> It's building up an array of advertised address in the order of pod ordinals (so redpanda-0 would be the first element and so forth) which is templated a second time at pod startup time with $SERVICE_NAME (which is NOT a service name, it's the pod name. [...]
> 
> This is again on the "to fix list" but it stable enough for now that we're leaving it as is to focus on less stable things.
> 
> tl;dr: expected behavior. If you have a more elaborate configuration those values could be different each time they're repeated.

### What did you expect to happen?

Seems like it should only list one instance of this per advertised address configured. In user's case, we believe it should only be once. 

### How can we reproduce it (as minimally and precisely as possible)?. Please include values file.

Can grab from user if necessary.

### Anything else we need to know?

Internal discussion: https://redpandadata.slack.com/archives/C01H6JRQX1S/p1717701360772929

### Which are the affected charts?

Redpanda

### Chart Version(s)

At least on 5.7.41 and likely later

### Cloud provider

self-hosted kubernetes
